pkg/store/bucket.go

```diff
@@ -3341,13 +3349,10 @@ func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, a
 		r.stats.ChunksFetchedSizeSum += units.Base2Bytes(len(*nb))
 		err = populateChunk(&(res[pIdx.seriesEntry].chks[pIdx.chunk]), rawChunk((*nb)[n:]), aggrs, r.save, calculateChunkChecksum)
 		if err != nil {
-			r.block.chunkPool.Put(nb)
 			return errors.Wrap(err, "populate chunk")
 		}
 		r.stats.chunksTouched++
 		r.stats.ChunksTouchedSizeSum += units.Base2Bytes(int(chunkDataLen))
-
-		r.block.chunkPool.Put(nb)
```

Comment:
Hm, where did this go?

Reply:
Idea was to put everything into the slice that is Put() back during Close() but it seems like this refactoring was faulty. I removed it.